diff --git a/src/context.rs b/src/context.rs
index 5fd111f..11b1b70 100644
--- a/src/context.rs
+++ b/src/context.rs
@@ -3040,7 +3040,7 @@ impl ContextWriter {
               }
             }
             RestorationFilter::Sgrproj{set, xqd} => {
-              match rs.plane[pli].lrf_type {
+              match rp.lrf_type {
                 RESTORE_SGRPROJ => {
                   symbol_with_update!(self, w, 1, &mut self.fc.lrf_sgrproj_cdf);
                 }
@@ -3071,7 +3071,7 @@ impl ContextWriter {
               }
             }
             RestorationFilter::Wiener{coeffs} => {
-              match rs.plane[pli].lrf_type {
+              match rp.lrf_type {
                 RESTORE_WIENER => {
                   symbol_with_update!(self, w, 1, &mut self.fc.lrf_wiener_cdf);
                 }
